r1cs/gadgets/std/src/groups/nonnative/tests.rs

```diff
@@ -0,0 +1,126 @@
+use algebra::{
+    fields::{
+        Field,
+        secp256k1::Fq as secp256k1Fq,
+        bn_382::Fr as BN382Fr,
+    },
+    curves::secp256k1::Secp256k1Parameters,
+    groups::Group,
+    UniformRand,
+    ToBits,
+};
+use r1cs_core::ConstraintSystem;
+use crate::{
+    alloc::AllocGadget,
+    bits::boolean::Boolean,
+    groups::{
+        GroupGadget,
+        nonnative::short_weierstrass_jacobian::GroupAffineNonNativeGadget,
+        test::group_test_with_incomplete_add,
+    },
+    test_constraint_system::TestConstraintSystem,
+};
+use rand::thread_rng;
+
+
+#[allow(dead_code)]
+pub(crate) fn mul_bits_test<
+    ConstraintF: Field,
+    G: Group,
+    GG: GroupGadget<G, ConstraintF, Value = G>,
+>()
```

Comment:
Why do we have an extra mul_bits_test, and do not use the one from groups/mod.rs?

Reply:
Too much resource intensive. This is a reduced version of that one.

Reply:
Good to know. Then let us add a comment.